### PR TITLE
docs(media): add an example for using a custom JSON encoder

### PR DESCRIPTION
**Pull request by maxking**
# Summary of Changes

Adds an example on how to override the default json encoder used by Falcon.

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [x] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

### CODE REVIEW

File: falcon/media/json.py [hunk 1]
```diff
@@ -94,6 +94,37 @@ class JSONHandler(BaseHandler):
             ),
         )
 
+    You can also override the JSONEncoder used by using a custom Encoder by
```
Comment (vytas7) [inline falcon/media/json.py:97]: Maybe we could rephrase this slightly to avoid the repetition of _used_ and _using_?

**Review by vytas7 — COMMENTED** [review]
Thanks for this improvement to our docs @maxking!

This is not critical to pass review, but while at it, could you also link to the new paragraph from this FAQ answer: [How can I use resp.media with types like datetime?](https://falcon.readthedocs.io/en/latest/user/faq.html#how-can-i-use-resp-media-with-types-like-datetime)

File: falcon/media/json.py (hunk 1 above)
Comment (maxking) [reply falcon/media/json.py]: Reworded the line.

**Comment by maxking** [comment]
Working on adding a reference back from FAQ, Sphinx doesn't seem to like sections inside the docstrings and I am trying to see how to create a reference without a section in a paragraph. Will update the PR as soon as I can figure that out. :D

File: falcon/media/json.py [hunk 2]
```diff
@@ -94,6 +94,37 @@ class JSONHandler(BaseHandler):
...
+        import falcon
+        from falcon import meda
+
+        class DatetimeEncoder(json.JSONEncoder):
+            \"\"\"Json Encoder that supports datetime objects.\"\"\"
+
+            def default(self, obj):
+                if isinstance(obj, datetime):
+                    return obj.isoformat()
+                super().default(obj)
```
Comment (CaselIT) [inline falcon/media/json.py:113]: could probably use return here, even if `JSONEncoder.default` will raise a type error

**Review by CaselIT — APPROVED** [review]

**Review by vytas7 — CHANGES_REQUESTED** [review]
This LGTM bar one typo.

And wrt linking to a target inside a docstring, it is apparently possible using some kind of marker that isn't a section (as sections are disallowed inside docstring), for instance, a `rubric`.

The following seems to work, but it adds visible rubrics that we don't use much elsewhere, so probably not worth the effort:

```diff
diff --git a/docs/user/faq.rst b/docs/user/faq.rst
index 9d3f09d..60c104f 100644
--- a/docs/user/faq.rst
+++ b/docs/user/faq.rst
@@ -1239,3 +1239,8 @@ Alternatively, you can set the Cookie header directly as demonstrated in this ve
 To include multiple values, simply use ``"; "`` to separate each name-value
 pair. For example, if you were to pass ``{'Cookie': 'xxx=yyy; hello=world'}``,
 you would get ``{'cookies': {'xxx': 'yyy', 'hello': 'world'}}``.
+
+How do I link to a rubric inside a docstring
+--------------------------------------------
+
+Trying... :ref:`target_to_paragraph`
diff --git a/falcon/media/json.py b/falcon/media/json.py
index e6cb0cf..a0359ae 100644
--- a/falcon/media/json.py
+++ b/falcon/media/json.py
@@ -179,6 +179,10 @@ class JSONHandlerWS(TextBaseHandlerWS):
         app = falcon.asgi.App()
         app.ws_options.media_handlers[falcon.WebSocketPayloadType.TEXT] = json_handler
 
+    .. _target_to_paragraph:
+
+    .. rubric:: About ensure_ascii
+
     By default, ``ensure_ascii`` is passed to the ``json.dumps`` function.
     If you override the ``dumps`` function, you will need to explicitly set
     ``ensure_ascii`` to ``False`` in order to enable the serialization of
```
```

File: docs/user/faq.rst [hunk 3]
```diff
@@ -915,6 +915,10 @@ Furthermore, different Internet media types such as YAML,
 types than JSON, either as part of the respective (de)serialization format, or
 via custom type extensions.
 
+.. seealso:: See :class:`falcon.media.json.JSONHandler` for an exmaple on how to
```
[the remark this answers is not in the document]
Comment (maxking) [reply docs/user/faq.rst]: Fixed.

**Comment by maxking** [comment]
Thanks @vytas7, updated the PR.

**Review by vytas7 — APPROVED** [review]
LGTM :+1:
(I've taken the liberty of tweaking those rubrics a bit to my liking.)

Thanks for this improvement to our docs, @maxking!

**Review by CaselIT — APPROVED** [review]

**Pull request merged**